include/xrpl/protocol/STAmount.h

```diff
@@ -76,104 +73,59 @@ class STAmount final : public STBase, public CountedObject<STAmount>
 
     // Max native value on network.
     static const std::uint64_t cMaxNativeN = 100000000000000000ull;
-    static const std::uint64_t cIssuedCurrency = 0x8000000000000000ull;
-    static const std::uint64_t cPositive = 0x4000000000000000ull;
-    static const std::uint64_t cMPToken = 0x2000000000000000ull;
-    static const std::uint64_t cValueMask = ~(cPositive | cMPToken);
+    static const std::uint64_t cNotNative = 0x8000000000000000ull;
+    static const std::uint64_t cPosNative = 0x4000000000000000ull;
 
     static std::uint64_t const uRateOne;
 
     //--------------------------------------------------------------------------
-    STAmount(SerialIter& sit, SField const& name);
+    STAmount(SerialIter& sit);
 
     struct unchecked
     {
         explicit unchecked() = default;
     };
 
-    // Do not call canonicalize
-    template <AssetType A>
-    STAmount(
-        SField const& name,
-        A const& asset,
-        mantissa_type mantissa,
-        exponent_type exponent,
-        bool native,
-        bool negative,
-        unchecked);
-
-    template <AssetType A>
     STAmount(
-        A const& asset,
+        Issue const& issue,
         mantissa_type mantissa,
         exponent_type exponent,
         bool native,
         bool negative,
         unchecked);
 
     // Call canonicalize
-    template <AssetType A>
     STAmount(
-        SField const& name,
-        A const& asset,
+        Issue const& issue,
         mantissa_type mantissa,
         exponent_type exponent,
         bool native,
         bool negative);
 
-    STAmount(SField const& name, std::int64_t mantissa);
-
-    STAmount(
-        SField const& name,
-        std::uint64_t mantissa = 0,
-        bool negative = false);
-
-    template <AssetType A>
-    STAmount(
-        SField const& name,
-        A const& asset,
-        std::uint64_t mantissa = 0,
-        int exponent = 0,
-        bool negative = false);
+    STAmount(std::int64_t mantissa, int64_tag_t);
```

Comment:
This needs a comment. I know the existing code is way undercommented, but new code should be better documented. (for this and other changes in this changeset, but this is the first one that caught my eye)